edx_exams/apps/api/v1/views.py

```diff
@@ -179,3 +179,78 @@ def patch(self, request, course_id):
             data = {"detail": "Invalid data", "errors": serializer.errors}
 
         return Response(status=response_status, data=data)
+
+
+class CourseExamConfigurationsView(APIView):
+    # todo: make parameters descriptions more useful.
+    """
+    View to create and update course exam configs for a specific course.
+
+    Given a course id and a proctoring provider, this view will either create a new course exam configuration
+    (if one doesn't exist), or modify the proctoring provider on an existing course exam config.
+
+    HTTP PATCH
+    Creates or updates a CourseExamConfiguration.
+    Expected PATCH data: {
+        'name': 'test_provider',
+        'verbose_name': 'testing_provider',
+        'lti_configuration_id': '123456789'
+    }
+    **PATCH data Parameters**
+        * name: This is the name of the proctoring provider.
+        * verbose_name: This is the verbose name of the proctoring provider.
+        * lti_configuration_id: This is the LTI configuration id of the proctoring provider.
+    **Exceptions**
+        * HTTP_400_BAD_REQUEST
+    """
+
+    authentication_classes = (JwtAuthentication,)
+    permission_classes = (StaffUserPermissions,)
+
+    @classmethod
+    def handle_config(cls, provider, course_id):
+        """
+        Helper method that decides whether to update existing or create new config.
+        """
+        # todo: is there a better way to check whether the config already exists?
+        # todo: does the course_id need a uniqueness constraint in the CourseExamConfiguration model?
+        config = CourseExamConfiguration.objects.filter(course_id=course_id)
+
+        # update if config exists
+        if len(config) == 1:
```

Comment:
[exists](https://docs.djangoproject.com/en/3.2/ref/models/querysets/#exists) is supposedly a little faster than checking for a non-empty QuerySet this way. But this is moot if you go with Alie's advice about the `update_or_create` method.

Reply:
This is really good to know, @MichaelRoytman! I ended up changing to `update_or_create` method but I'll plan on using exists moving forward if needed. :)